lib/wanda/notes.md

```diff
@@ -0,0 +1,40 @@
+  Scenario: we have selected some checks for the hosts of a given cluster and the user triggers an execution
+
+  what does it mean "requesting an execution for the selection"?
+  could it be something like the following?
+
+  starting the process of gathering facts form targets and evaluating expectations **for all the hosts** on a given cluster
+  and recalculate the health of the single hosts and of the cluster as a whole given the checks results
+
+  is it a usecase to start execution for the check selection of a **single host**?
+
+  information at hand:
+  - an identifier for the currently triggered execution (by the user or by time passing)
+  - cluster id
+  - list of the target hosts (aka agents) of the cluster
+  - list of the selected checks (12ER4, FD54R, ...) - each agent has its own selection
+  - provider (atm this is the cluster's provided - which is simply the discovered cluster of a host...)
+
+  Assumptions:
+  - the selected checks may vary per target agent
```

Comment:
You mean within the same cluster?

Reply:
Yes.

As of now what we call a CheckSelection is a list of checks that we want to execute on a target Cluster and the same selection would be applied to all the hosts of that cluster.

However, the user needs the flexibility to configure a more fine grained CheckSelection in order to better suite the complexity of checking SAP infrastructures.
So we're going in the direction where each host has its own Check Selection.